crates/bevy_remote/src/http.rs

```diff
@@ -36,6 +37,37 @@ pub const DEFAULT_PORT: u16 = 15702;
 /// The default host address that Bevy will use for its server.
 pub const DEFAULT_ADDR: IpAddr = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
 
+/// A struct that holds a collection of HTTP headers.
+///
+/// This struct is used to store a set of HTTP headers as key-value pairs, where the keys are
+/// of type [`HeaderName`] and the values are of type [`HeaderValue`].
+///
+#[derive(Debug, Resource, Clone)]
+pub struct Headers {
+    headers: HashMap<HeaderName, HeaderValue>,
+}
+
+impl Headers {
+    /// Create a new instance of `Headers`.
+    pub fn new() -> Self {
+        Self {
+            headers: HashMap::new(),
+        }
+    }
+
+    /// Add a key value pair to the `Headers` instance.
+    pub fn add(mut self, key: HeaderName, value: HeaderValue) -> Self {
```

Comment:
I'd rather this be called `insert` as that is more common for maps which override previous values based on key. Not going to block on this though.